geoportailv3/static/js/map/mapdirective.js

```diff
@@ -26,7 +26,8 @@ goog.require('ol.proj');
 app.mapDirective = function(appMapTemplateUrl) {
   return {
     scope: {
-      'map': '=appMapMap'
+      'map': '=appMapMap',
+      'mainctrl': '=appMapMainctrl'
```

Comment:
Instead of passing the mainCtrl here, you can add a controller to the Toggle3dDirective below and inject the `ngeoOlcsService`.

This service contains the manager.
When the manager is null (service not initialized or 3D not allowed), the component should not be visible.

In order to handle extra behaviour of toggle3d you can override the `toggle3d` method of app.MainController.Lux3DManager (no need for firing events, as evoked during the daily).